src/runtime/auto-prompt-manager.js

```diff
@@ -502,37 +502,40 @@ export class AutoPromptManager {
       autoPromptType === AutoPromptType.CONTRIBUTION ||
       autoPromptType === AutoPromptType.CONTRIBUTION_LARGE
     ) {
-      let preferSurveyOverContributionPrompt = await this.isExperimentEnabled_(
+      // Survery take highest priority if this flag is enabled.
+      const prioritizeSurvey = await this.isExperimentEnabled_(
         article,
         ExperimentFlags.SURVEY_TRIGGERING_PRIORITY
       );
 
-      // If audienceActions contains both contribution and survey actions,
-      // we will use the order from audienceActions to determine the preference.
-      const contributionIndex = potentialActions.findIndex(
+      let previousPrompts = [];
+      if (dismissedPrompts) {
+        previousPrompts = dismissedPrompts.split(',');
+        previousPrompts = previousPrompts.map((action) =>
```

Comment:
Could we combine this line with the line above?
```suggestion
         previousPrompts.map((action) =>
```

Reply:
Ex: `previousPrompts = dismissedPrompts.split(',').map(...`

Reply:
Done